### PR TITLE
External volume support

### PR DESCRIPTION
**Pull request by afraisse**
Support for external volumes with a bit of refactoring (made Volume class abstract).

### CODE REVIEW

**Comment by bosschaert** [comment]
Just wondering why this pull request still isn't merged? Is there any issue with it? @colin-msphere has created a unit test for it here: https://github.com/afraisse/marathon-client/pull/1

I think it would be great if this could be merged as external volume support is sorely missing from the current release...

**Comment by afraisse** [comment]
Thanks @colin-msphere for the unit testing btw ;)

**Comment by bosschaert** [comment]
I'm trying this branch out to create the following external volume definition. It's a bit strange because the API surfaces a method called getExternalVolumeInfo() but that returns a package protected class, so that doesn't sound right...

How would I model a definition that will look in Marathon like this:

```"volumes": [
      {
        "containerPath": "/somewhere/in/the/conainer",
        "external": {
          "name": "my-persistent",
          "provider": "dvdi",
          "options": {
            "dvdi/driver": "rexray"
          }
        },
        "mode": "RW"
      }
```

**Comment by afraisse** [comment]
The getters/setters on externalVolumeInfo are just for means of serialization. There are proxies at the ExternalVolume class level that you can use.
```
ExternalVolume externalVol = new ExternalVolume();
externalVol.setContainerPath("/somewhere/in/the/container");
externalVol.setName("my-persistent");
externalVol.setProvider("dvdi"); // DVDI is default btw
externalVol.setDriver("rexray");
externalVol.setMode("RW");
```

**Comment by bosschaert** [comment]
@afraisse got that to work - thanks.

Still it looks like the getExternalVolumeInfo()/setExternalVolumeInfo() methods should not be public since they return a non-public class...

**Comment by afraisse** [comment]
You are right indeed, thanks.

File: src/test/java/mesosphere/marathon/client/model/v2/IntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import mesosphere.marathon.client.Marathon;
+import mesosphere.marathon.client.MarathonClient;
+import mesosphere.marathon.client.utils.MarathonException;
+
+@Ignore
+public class IntegrationTest {
```
Comment (colin-msphere) [inline src/test/java/mesosphere/marathon/client/model/v2/IntegrationTest.java]: I think this test/file/class can be removed in favor of the added unit test using `MockWebServer`.

**Pull request merged**